src/database/queries.rs

```diff
@@ -117,84 +154,110 @@ pub(crate) async fn blocks_storage_intersection(
         FROM blocks
         WHERE id = $1
         ",
+        id
     )
-    .bind(id)
     .fetch_one(conn)
     .await
     .map_err(Into::into)
 }
 
+/// Get a block by block number from the relational database
 #[cfg(test)]
 pub(crate) async fn get_full_block_by_num(
     conn: &mut sqlx::PgConnection,
     block_num: u32,
 ) -> Result<BlockModel> {
-    sqlx::query_as(
+    let safe_block_num = i32::try_from(block_num).unwrap_or(i32::MAX);
+    sqlx::query_as!(
+        BlockModel,
         "
-        SELECT parent_hash, hash, block_num, state_root, extrinsics_root, digest, ext, spec
+        SELECT id, parent_hash, hash, block_num, state_root, extrinsics_root, digest, ext, spec
         FROM blocks
         WHERE block_num = $1
         ",
+        safe_block_num
     )
-    .bind(block_num as i32)
     .fetch_one(conn)
     .await
     .map_err(Into::into)
 }
 
-/// check if a runtime versioned metadata exists in the database
+/// Check if the runtime version identified by `spec` exists in the relational database
 pub(crate) async fn check_if_meta_exists(spec: u32, conn: &mut PgConnection) -> Result<bool> {
-    let row: (bool,) =
-        sqlx::query_as(r#"SELECT EXISTS(SELECT version FROM metadata WHERE version = $1)"#)
-            .bind(spec)
-            .fetch_one(conn)
-            .await?;
-    Ok(row.0)
+    let safe_spec = i32::try_from(spec).unwrap_or(i32::MAX);
+    let does_exist = sqlx::query_as!(
+        DoesExist,
+        r#"SELECT EXISTS(SELECT version FROM metadata WHERE version = $1)"#,
+        safe_spec
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(does_exist.exists.unwrap_or(false))
 }
 
+/// Check if the block identified by `hash` exists in the relational database
 pub(crate) async fn has_block<B: BlockT>(hash: B::Hash, conn: &mut PgConnection) -> Result<bool> {
     let hash = hash.as_ref();
-    let row: (bool,) = sqlx::query_as(r#"SELECT EXISTS(SELECT 1 FROM blocks WHERE hash = $1)"#)
-        .bind(hash)
-        .fetch_one(conn)
-        .await?;
-    Ok(row.0)
+    let does_exist = sqlx::query_as!(
+        DoesExist,
+        r#"SELECT EXISTS(SELECT 1 FROM blocks WHERE hash = $1)"#,
+        hash,
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(does_exist.exists.unwrap_or(false))
 }
 
-/// Returns a list of block_numbers, out of the passed-in blocknumbers, which exist in the database
+/// Get a list of block_numbers, out of the passed-in blocknumbers, which exist in the relational
+/// database
 pub(crate) async fn has_blocks<B: BlockT>(
     nums: &[u32],
     conn: &mut PgConnection,
 ) -> Result<Vec<u32>> {
-    let query = String::from("SELECT block_num FROM blocks WHERE block_num = ANY ($1)");
-    let row = sqlx::query_as::<_, (i32,)>(query.as_str())
-        .bind(nums)
-        .fetch_all(conn)
-        .await?;
-    Ok(row.into_iter().map(|r| r.0 as u32).collect())
+    let safe_nums: Vec<i32> = nums
+        .iter()
+        .map(|n| i32::try_from(*n).unwrap_or(i32::MAX))
```

Comment:
You should be able to `filter` or `filter_map` here I think, to avoid keeping bad input as `i32::MAX`.